langchain4j-bedrock/src/test/java/dev/langchain4j/model/bedrock/BedrockEmbeddingIT.java

```diff
@@ -83,4 +83,70 @@ void testBedrockTitanEmbeddingModelV2() {
         assertThat(embeddingModel.dimension()).isEqualTo(256);
     }
 
+    @Test
+    void testBedrockCohereEmbedEnglishTextV3() {
+
+        BedrockCohereEmbeddingModel embeddingModel = BedrockCohereEmbeddingModel
+                .builder()
+                .region(Region.US_EAST_1)
+                .maxRetries(1)
+                .model(BedrockCohereEmbeddingModel.Types.CohereEmbedEnglishTextV3.getValue())
+                .build();
+
+        assertThat(embeddingModel).isNotNull();
+
+        List<TextSegment> segments = Collections.singletonList(TextSegment.from("one"));
+
+        Response<List<Embedding>> response = embeddingModel.embedAll(segments);
+        assertThat(response).isNotNull();
+
+        List<Embedding> embeddings = response.content();
+        assertThat(embeddings).hasSize(1);
+
+        Embedding embedding = embeddings.get(0);
+        assertThat(embedding.vector()).hasSize(1024);
+
+        TokenUsage tokenUsage = response.tokenUsage();
+        assertThat(tokenUsage.inputTokenCount()).isEqualTo(0);
+        assertThat(tokenUsage.outputTokenCount()).isNull();
+        assertThat(tokenUsage.totalTokenCount()).isEqualTo(0);
+
+        assertThat(response.finishReason()).isNull();
+
+        assertThat(embeddingModel.dimension()).isEqualTo(1024);
+    }
+
+    @Test
+    void testBedrockCohereEmbedMultilingualTextV3() {
+
+        BedrockCohereEmbeddingModel embeddingModel = BedrockCohereEmbeddingModel
+                .builder()
+                .region(Region.US_EAST_1)
+                .maxRetries(1)
+                .model(BedrockCohereEmbeddingModel.Types.CohereEmbedMultilingualTextV3.getValue())
+                .build();
+
+        assertThat(embeddingModel).isNotNull();
+
+        List<TextSegment> segments = Collections.singletonList(TextSegment.from("one"));
```

Comment:
this does not test multiple embeddings

Reply:
Changed. Added new test for multiple TextSegments 
![Uploading Screenshot 2024-11-08 at 15.22.52.png…]()